src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -120,9 +121,12 @@ public StreamingResponseBody streamEvents(
                 }
 
                 // check that offsets are not out of bounds and partitions exist
-                if (cursors != null && !topicRepository.areCursorsValid(topic, cursors)) {
-                    writeProblemResponse(response, outputStream, PRECONDITION_FAILED, "cursors are not valid");
-                    return;
+                if (cursors != null) {
+                    final Optional<String> errorMessage = topicRepository.validateCursors(topic, cursors);
```

Comment:
Well, actually no. My suggestion is to remove `areCursorsValid` method from topic repository interface, create Exception type `CursorInvalidException` with property reason, and validate cursor while creating consumers, throwing exceptions sometimes.